litt/littdb_config.go

```diff
@@ -0,0 +1,196 @@
+	MetricsEnabled bool
+
+	// The namespace to use for metrics. If empty, the default namespace "litt" is used.
+	MetricsNamespace string
+
+	// The prometheus registry to use for metrics. If nil and metrics are enabled, a new registry is created.
+	MetricsRegistry *prometheus.Registry
+
+	// The port to use for the metrics server. Ignored if MetricsEnabled is false or MetricsRegistry is not nil.
+	MetricsPort int
+
+	// The interval at which various DB metrics are updated. The default is 1 second.
+	MetricsUpdateInterval time.Duration
+}
+
+// DefaultConfig returns a Config with default values.
+func DefaultConfig(paths ...string) (*Config, error) {
+	if len(paths) == 0 {
+		return nil, fmt.Errorf("at least one path must be provided")
+	}
+
+	seed := time.Now().UnixNano()
+	saltShaker := rand.New(rand.NewSource(seed))
+
+	loggerConfig := common.DefaultLoggerConfig()
+
+	return &Config{
+		CTX:                      context.Background(),
+		Paths:                    paths,
+		LoggerConfig:             &loggerConfig,
+		TimeSource:               time.Now,
+		GCPeriod:                 5 * time.Minute,
+		GCBatchSize:              10_000,
+		ShardingFactor:           8,
+		SaltShaker:               saltShaker,
+		KeymapType:               keymap.LevelDBKeymapType,
+		ControlChannelSize:       64,
+		TargetSegmentFileSize:    math.MaxUint32,
+		MaxSegmentKeyCount:       50_000,
+		TargetSegmentKeyFileSize: 2 * units.MiB,
+		Fsync:                    true,
+		DoubleWriteProtection:    false,
+		MetricsEnabled:           false,
+		MetricsNamespace:         "litt",
+		MetricsPort:              8080,
+		MetricsUpdateInterval:    time.Second,
+	}, nil
+}
+
+// SanityCheck performs a sanity check on the configuration, returning an error if any of the configuration
+// settings are invalid. The config returned by DefaultConfig() is guaranteed to pass this check if unmodified.
+func (c *Config) SanityCheck() error {
+	if c.CTX == nil {
+		return fmt.Errorf("context cannot be nil")
+	}
+	if c.Paths == nil || len(c.Paths) == 0 {
+		return fmt.Errorf("at least one path must be provided")
+	}
+	if c.Logger == nil && c.LoggerConfig == nil {
+		return fmt.Errorf("logger or logger config must be provided")
+	}
+	if c.TimeSource == nil {
+		return fmt.Errorf("time source cannot be nil")
+	}
+	if c.GCBatchSize == 0 {
+		return fmt.Errorf("gc batch size must be at least 1")
+	}
+	if c.ShardingFactor == 0 {
+		return fmt.Errorf("sharding factor must be at least 1")
+	}
+	if c.ControlChannelSize == 0 {
+		return fmt.Errorf("control channel size must be at least 1")
+	}
+	if c.TargetSegmentFileSize == 0 {
+		return fmt.Errorf("target segment file size must be at least 1")
+	}
+	if c.MaxSegmentKeyCount == 0 {
+		return fmt.Errorf("max segment key count must be at least 1")
+	}
+	if c.TargetSegmentKeyFileSize == 0 {
+		return fmt.Errorf("target segment key file size must be at least 1")
+	}
+	if c.GCPeriod == 0 {
+		return fmt.Errorf("gc period must be at least 1")
```

Comment:
a bit odd to say duration must be "at least 1". Consider

"gc period must be greater than 0"

Reply:
updated wording